framework/src/actions/SetupMeshAction.C

```diff
@@ -167,6 +167,25 @@ SetupMeshAction::act()
   // Create the mesh object and tell it to build itself
   if (_current_task == "setup_mesh")
   {
+    /**
+     * MOOSE has long supported a legacy syntax where the [Mesh] block doesn't require a "type" if
+     * the user wants a FileMesh type. This behavior combined with the fact that "setup_mesh" is a
+     * required task, means that the SetupMeshAction might be auto-built by the Parser to satisfy
+     * that task if no other Action registers to satisfy that task. However, we'd like to avoid
+     * requiring that users include a [Mesh] block if they are generating the mesh through the
+     * [MeshGeneraotrs] system. In the case where only the [MeshGenerators] block exists in the
+     * input file, this Action would have been constructed, but the "type" would be set to the
+     * (wrong) default (e.g. FileMesh). We'll attempt to detect that case by seeing if this task
+     * was AutoBuilt or not. If it is, we'll quietly switch the type and give it an empty
+     * InputParameters object to "properly" attempt to construct the right MooseMesh type when
+     * the "setup_mesh" task executes to handle mesh generators.
+     */
```

Comment:
@permcody In what cases where you have a Mesh and MeshGenerator block should the type in Mesh be anything different than MeshGeneratorMesh? As far as I know you can't run for example a TransformGenerator on a FileMesh, it would have to be on the result of a FileMeshGenerator.  

The reason I ask is because a use case that I can see where you would always need a Mesh block even if you created the mesh through MeshGenerators is when you have Contact or need the displaced mesh:

```
[MeshGenerators] # Convert CUBIT geometry (built in 100x inches) into meters.
  [file_mesh]
    type = FileMeshGenerator
    file = my_mesh.e
  []
[]

[Mesh]
  type = MeshGeneratorMesh <---- I shouldn't need to specify this in this case because I have a MeshGenerator block 
  patch_size = 5 
  patch_update_strategy = auto
  displacements = 'disp_x disp_y'
[]
```
I think the type in Mesh should always be set to MeshGeneratorMesh any time you have a MeshGenerators block, unless I am missing some other use case where you could have both.  The fact that in many of our problems (mechanics, contact) would necessitate the need for both blocks anyways gives my vote to nest the MeshGenerators block inside the Mesh block so you could do something like:

```
[Mesh]
  patch_size = 5 
  patch_update_strategy = auto
  displacements = 'disp_x disp_y'
  [file_mesh]
    type = FileMeshGenerator
    file = my_mesh.e
  []
[]
```

Reply:
Yeah - I'm abandoning this logic. We will force the type to switch to MeshGeneratorMesh in the case of having Generators.

I think the idea is that we could have a FileMesh and Generators and switch back and forth between them - we aren't going to do that now.

Reply:
If you are going to have a FileMesh and Generators why not just read it in through a FileMeshGenerator in addition to the other Generators? Then through MeshGeneratorMesh it should provide the same behavior - I know we aren't going to do that now, but I am just trying to understand the use case.

Reply:
lol, this accidentally is what I proposed, use `[Mesh]` but not `[MeshGenerators]`.

Reply:
@gambka - That's exactly what we'll do eventually.

Reply:
@permcody.  I must be missing something.  Doesn't the functionality already exist? There is already an existing `FileMeshGenerator` and I've used it with other `MeshGenerators`.   The only thing missing is the current requirement of having to specifying `type = MeshGeneratorMesh` in the `[Mesh]` block currently, which is what this PR is addressing with your new logic of forcing the type to switch to MeshGeneratorMesh any time we have Generators.

Reply:
You aren't missing anything. This PR only does two things:

1) Moves the syntax into Mesh
2) Removes the requirement for having an explicit type for either FileMesh (no generators and no type specified) or MeshGeneratedMesh (generators and no type specified). 

It also throws a warning if the Mesh block tries to build something else beside the MeshGeneratorMesh when there are generators.